### PR TITLE
[RHCLOUD-24017] Adding sort and dir parameters

### PR DESCRIPTION
**Pull request by justinorringer**
### Changes
- Passing the sort and dir parameters with the PaginationCtx (we've already talked about how this design may need changes).
- Sort by and dir parameters as in the [ADR](https://docs.google.com/document/d/1YU7RC-ij-LaGW4HG1M6kVD4YzbXsLnF7Ex56z9MnHaY/edit#heading=h.vvf01oumsj9r).
- Default sort is created, default dir is asc.

For doing the actual sorting, can you all think of an alternative to creating a custom sort function and using sort.Slice() ?

### CODE REVIEW

**Comment by justinorringer** [comment]
Before, the service was handling pagination in memory (which would cause OOM exceptions in prod), so GORM now handles that logic in addition to sort by and direction parameters.

File: exports/exports_test.go [hunk 1]
```diff
@@ -199,6 +200,205 @@ var _ = Describe("The public API", func() {
...
+
+		router.ServeHTTP(rr, req)
+
+		Expect(rr.Code).To(Equal(http.StatusOK))
+		Expect(rr.Body.String()).To(ContainSubstring(fmt.Sprintf("count\":%d", count)))
+		Expect(rr.Body.String()).To(ContainSubstring("Test Export Request 111"))
+		Expect(rr.Body.String()).To(ContainSubstring("Test Export Request 120"))
+
+		// ----------------
+		// all the data
```
Comment (CodyWMitchell) [inline exports/exports_test.go:252]: Nit-picky, but we may be able to clean up some of the duplication here?

**Review by CodyWMitchell — APPROVED** [review]
LGTM!

**Review by CodyWMitchell — APPROVED** [review]
LGTM!

**Pull request merged**